### ISSUE TITLE
Failed to initialize CoreCLR, HRESULT: 0x80004005 with daily build

### ISSUE THREAD

**Issue by redknightlois** [dotnet/core-setup#4016]
## Steps to reproduce

I had been having this issue too. I removed CoreCLR entirely (also deleted coreclr and nuget cached packages) and was able to repro on a clean install.

Install SDK 2.1.104 (x64) from: ```https://www.microsoft.com/net/download/thank-you/dotnet-sdk-2.1.104-windows-x64-installer```
Run ```dotnet --version``` it should return: ```2.1.104```
Install SDK 2.1.300-preview3-008443 from ```https://dotnetcli.blob.core.windows.net/dotnet/Sdk/master/dotnet-sdk-latest-win-x64.exe```
Run ```dotnet --version```

![image](https://user-images.githubusercontent.com/1045411/38635294-4b96e3f4-3d9b-11e8-918b-723b07325351.png)

## Actual behavior

Error with: ```Failed to initialize CoreCLR, HRESULT: 0x80004005```

cc @karelz 

PS: Running on the bleeding edge 'daily fixes on intrinsics stuff'

**Comment by redknightlois** [comment]
No exceptions at all, it just goes down. VS isn't able to show me anything either because the process doesn't even start loading the .dlls ... My bet is that there has been some changes to the CLI from around 26403-01 to today. I was able to work with the latest until I upgraded the bits to the daily today.

**Comment by redknightlois** [comment]
I run it under windbg. The runtime is able to be loaded but eventually, an access violation (first chance) is thrown. The suspect is VTune, but there is no much information that I can get without some extra hints on how to proceed forward with some tracing information. 

```
ModLoad: 00007ffa`05e90000 00007ffa`05f8a000   C:\WINDOWS\System32\ucrtbase.dll
(13b4.2f94): Break instruction exception - code 80000003 (first chance)
ntdll!LdrpDoDebuggerBreak+0x30:
00007ffa`0933cf6c cc              int     3
0:000> g
ModLoad: 00007ff9`de960000 00007ff9`de9c4000   C:\Program Files\dotnet\host\fxr\2.1.0-preview2-26406-04\hostfxr.dll
ModLoad: 00007ff9`d6c60000 00007ff9`d6cf1000   C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.1.0-preview2-26406-04\hostpolicy.dll
ModLoad: 00007ff9`a4c70000 00007ff9`a5235000   C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.1.0-preview2-26406-04\coreclr.dll
ModLoad: 00007ffa`07510000 00007ffa`075b1000   C:\WINDOWS\System32\ADVAPI32.dll
ModLoad: 00007ffa`079c0000 00007ffa`07a5e000   C:\WINDOWS\System32\msvcrt.dll
ModLoad: 00007ffa`08ea0000 00007ffa`08efb000   C:\WINDOWS\System32\sechost.dll
ModLoad: 00007ffa`090a0000 00007ffa`091c4000   C:\WINDOWS\System32\RPCRT4.dll
ModLoad: 00007ffa`075c0000 00007ffa`07711000   C:\WINDOWS\System32\ole32.dll
ModLoad: 00007ffa`07100000 00007ffa`07423000   C:\WINDOWS\System32\combase.dll
ModLoad: 00007ffa`05640000 00007ffa`056ba000   C:\WINDOWS\System32\bcryptPrimitives.dll
ModLoad: 00007ffa`07460000 00007ffa`07488000   C:\WINDOWS\System32\GDI32.dll
ModLoad: 00007ffa`05cd0000 00007ffa`05e62000   C:\WINDOWS\System32\gdi32full.dll
ModLoad: 00007ffa`05900000 00007ffa`0599f000   C:\WINDOWS\System32\msvcp_win.dll
ModLoad: 00007ffa`08f00000 00007ffa`09090000   C:\WINDOWS\System32\USER32.dll
ModLoad: 00007ffa`05e70000 00007ffa`05e90000   C:\WINDOWS\System32\win32u.dll
ModLoad: 00007ffa`06f70000 00007ffa`07032000   C:\WINDOWS\System32\OLEAUT32.dll
ModLoad: 00007ffa`077a0000 00007ffa`077f1000   C:\WINDOWS\System32\SHLWAPI.dll
ModLoad: 00007ff9`ff8d0000 00007ff9`ff8da000   C:\WINDOWS\SYSTEM32\VERSION.dll
ModLoad: 00007ffa`050e0000 00007ffa`05105000   C:\WINDOWS\SYSTEM32\bcrypt.dll
ModLoad: 00007ffa`07430000 00007ffa`0745d000   C:\WINDOWS\System32\IMM32.DLL
onecore\windows\core\console\open\src\host\srvinit.cpp(411)\conhost.exe!00007FF6C331C372: (caller: 00007FF6C3318CD9) ReturnHr(4) tid(2c7c) 80070032 The request is not supported.
(13b4.2f94): Unknown exception - code 04242420 (first chance)
onecore\windows\core\console\open\src\host\srvinit.cpp(411)\conhost.exe!00007FF6C331C372: (caller: 00007FF6C3318CD9) ReturnHr(5) tid(2c7c) 80070032 The request is not supported.
ModLoad: 00007ff9`a5510000 00007ff9`a567d000   C:\Program Files (x86)\IntelSWTools\VTune Amplifier 2018\bin64\amplxe_samplingmrte_clrprof_1.0.dll
ModLoad: 00007ffa`07730000 00007ffa`0779c000   C:\WINDOWS\System32\WS2_32.dll
(13b4.2f94): Access violation - code c0000005 (first chance)
First chance exceptions are reported before any exception handling.
This exception may be expected and handled.
*** ERROR: Symbol file could not be found.  Defaulted to export symbols for C:\Program Files (x86)\IntelSWTools\VTune Amplifier 2018\bin64\amplxe_samplingmrte_clrprof_1.0.dll - 
amplxe_samplingmrte_clrprof_1_0!DllUnregisterServer+0x2bfec:
00007ff9`a553d5f0 488991e0000000  mov     qword ptr [rcx+0E0h],rdx ds:00000000`000000e0=????????????????
0:000> g
ModLoad: 00007ffa`055b0000 00007ffa`055c1000   C:\WINDOWS\System32\kernel.appcore.dll
ntdll!NtTerminateProcess+0x14:
00007ffa`0930a524 c3              ret
0:000> g
onecore\windows\core\console\open\src\server\devicecomm.cpp(153)\conhost.exe!00007FF6C331A79A: (caller: 00007FFA07053034) ReturnHr(6) tid(2c7c) 800700E9 No process is on the other end of the pipe.
ntdll!NtWaitForWorkViaWorkerFactory+0x14:
00007ffa`0930d8f4 c3              ret
```

**Comment by karelz** [comment]
IMO it is a problem in VTune, not in CoreCLR/.NET Core. The AV comes from VTune after all. I don't think VTune works with our profiling APIs.

CoreCLR repo is the right place to ask about Profiling APIs, and what they do under the hood.

**Comment by redknightlois** [comment]
Can you move the issue there?

**Comment by karelz** [comment]
I don't think we should move it. I don't believe there is anything actionable for CoreCLR.
VTune just can't handle CoreCLR profiler generated code IMO. Which blocks VTune investigations while CoreCLR profiler is enabled ... likely not a huge problem with a reasonable workaround. It impacts only some rare developer scenarios.

cc @noahfalk if he has different opinion.

I would suggest to close this issue.

**Comment by redknightlois** [comment]
The problem is that it used to work, and I am one of those of those rare developers. This is a new development with the latest runtimes. I have been looking at the issues and PRs and there are a few potential cases out there that touched related places in the code. I can pinpoint the issue and figure out where it was introduced and then we can decide who is at fault. What I would need is the ability to download specific versions of the runtime. Is there any place I could get them?

**Comment by fiigii** [comment]
> The issue is with VTune 2018. As soon as I disabled the CORECLR_ENABLE_PROFILING flag it runs. 

If you are using VTune 2018 update1/2, could you try to uninstall it and use  VTune 2018 instead to see if it happen also? We are working with the VTune team to figure it out.

**Comment by fiigii** [comment]
I can reproduce this crash with the latest .NET Core SDK (2.1.3-preview3) when using VTune 2018 update 2. 

However, this VTune version works fine with the previous versions of CoreCLR (e.g., 2.1.104), this may be a .NET Core issue also. Can any expert help to look into it? 

We will follow up with VTune team for potential VTune issue.

**Comment by noahfalk** [comment]
You can start with me from the CoreCLR side. I'd need a crash dump or some instructions how to repro (and I don't have VTune already so I'd have to figure out how to get a copy). It looks like the AV occurs in VTune's code, but if that is caused by bad data/unexpected call that came from CoreCLR it will be helpful for someone who understands the VTune code and has access to symbols to help connect those dots.

**Comment by redknightlois** [comment]
@noahfalk Probably MS has it in its internal download license service, if not you can get it for free from Intel's own site with a 90 days renewable license. https://software.intel.com/en-us/system-studio/choose-download. 

The repro is as simple as following the instructions with a clean install. You can probably install the latest SDK and make it fail without starting from a clean setup, but that is how I reproduced it to fail with the ```dotnet --version``` command.

**Comment by redknightlois** [comment]
> If you are using VTune 2018 update1/2, could you try to uninstall it and use VTune 2018 instead to see if it happen also? We are working with the VTune team to figure it out.

@fiigii Not really VTune older than Update 2 crashes because of the Windows Kernel Memory Isolation mitigation for Meltdown/Spectre. Won't ensure it will work (because it doesn't), but will probably allow us to figure out if CoreCLR would crash though.

**Comment by karelz** [comment]
@fiigii can someone from VTune at least point @noahfalk in the right direction? Debugging tools like VTune without symbols and source code to find out what it doesn't like is pretty difficult and time consuming :(

@noahfalk would it make sense to move the issue to CoreCLR (or create a new one there)?

**Comment by fiigii** [comment]
@redknightlois Are you using Advanced Hotspot Analysis? If yes, I think VTune 2018 update 2 does not need the two environment variables (CORECLR_ENABLE_PROFILING & CORECLR_PROFILER). So just remove/disable these two variables can temporarily solve this problem. I have trived on my machine.

**Comment by redknightlois** [comment]
@fiigii No, I am using general exploration and Microarchitecture analysis to do fine measurements over tight loops.

**Comment by fiigii** [comment]
Advanced Hotspot Analysis for .NET Core app is officially supported in VTune 2018 update 2, so it does not require the two environment variable. I did not see general exploration support in the latest release note, but **I tried that VTune 2018 update 2 works fine with general exploration analysis for .NET Core 2.1 (without the two variables)**. So, disabling these two variables also help to general exploration.

@karelz I think we can close this issue.

**Comment by fiigii** [comment]
@redknightlois The [.NET blog](https://blogs.msdn.microsoft.com/dotnet/2017/10/23/net-core-performance-profiling-with-intel-vtune-amplifier-2018/) and [Intel VTune cookbook](https://software.intel.com/en-us/vtune-amplifier-cookbook-profiling-a-net-core-application) have been updated to reflect the change of VTune 2018 update 2. Thank you for reporting this issue.
